agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1019,7 +1019,22 @@ bool backhaul_manager::backhaul_fsm_main(bool &skip_select)
             if (elapsed_time_s >= ap_metrics_reporting_info.reporting_interval_s) {
                 ap_metrics_reporting_info.last_reporting_time_point = now;
 
-                // TODO: report AP metrics
+                std::vector<sMacAddr> bssid_list;
+                for (auto &socket : slaves_sockets) {
+                    if (socket) {
+                        for (int i = 0; i < beerocks::IFACE_TOTAL_VAPS; ++i) {
+                            if (socket->vaps_list.vaps[i].mac != network_utils::ZERO_MAC) {
+                                bssid_list.push_back(socket->vaps_list.vaps[i].mac);
+                            }
+                        }
+                    }
+                }
+                if (!bssid_list.empty()) {
+                    send_slave_ap_metric_query_message(0, bssid_list);
```

Comment:
I think this deserves a comment to explain the 0:
```suggestion
                    // We must generate a new MID for the response, so set MID to 0 here.
                    send_slave_ap_metric_query_message(0, bssid_list);
```

Reply:
Added